resolver/resolver.go

```diff
@@ -118,6 +122,9 @@ func (r *Resolver) ResolveToLastNode(ctx context.Context, fpath path.Path) (cid.
 
 // ResolvePath fetches the node for given path. It returns the last item
 // returned by ResolvePathComponents and the last link traversed which can be used to recover the block.
+//
+// Note: if/when the context is cancelled or expires then if a multi-block ADL node is returned then it may not be
+// possible to load certain values.
```

Comment:
ok, so I've removed the other 3 timeouts but left this one in as it's at least plausible.

I'm still concerned that the other functions all return nodes where cancelling the context prevents the nodes from being traversable. However, the caller is able to control if/when this happens so it's less of a problem. I added a comment calling out the difficulty.

cc @warpfork any better ideas?